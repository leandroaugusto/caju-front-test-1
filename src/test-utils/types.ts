export interface ITestUtils {
  children?: React.ReactNode
}
