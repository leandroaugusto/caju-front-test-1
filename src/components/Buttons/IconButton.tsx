import styled from "styled-components";

import { pxToRem } from "~/utils/pxToRem";

export const IconButtonStyled = styled.button`
  cursor: pointer;
  border: ${pxToRem(2)} solid #64a98c;
  width: fit-content;
  padding: 0.25rem;
  border-radius: 1.5rem;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: transparent;
  svg {
    color: #64a98c;
  }
`;

type IconButtonProps = {
  children?: React.ReactNode;
} & React.HTMLAttributes<HTMLButtonElement>;

export const IconButton = (props: IconButtonProps) => {
  return <IconButtonStyled {...props}>{props.children}</IconButtonStyled>;
};
