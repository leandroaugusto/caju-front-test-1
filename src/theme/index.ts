const theme = {
  colors: {
    primary: "#e80537",
    secondary: "#4242DF",
    tertiary: "#9be59b",
    background: "#f9f9f9",
    buttons: {
      review: "#ff8858",
      approved: "#9be59b",
      reproved: "#ff919a"
    },
    columns: {
      review: {
        background: "#FDF8E9",
        title: "#EFC24D",
      },
      approved: {
        background: "#EEEEFD",
        title: "#4242DF",
      },
      reproved: {
        background: "#FBEDF6",
        title: "#CE2893",
      },
    }
  },
};

export default theme;
