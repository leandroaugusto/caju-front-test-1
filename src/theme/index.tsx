const theme = {
  colors: {
    review: {
      background: "#FDF8E9",
      title: "#EFC24D",
      button: "#ff8858",
    },
    approved: {
      background: "#EEEEFD",
      title: "#4242DF",
      button: "#9be59b",
    },
    reproved: {
      background: "#FBEDF6",
      title: "#CE2893",
      button: "#ff919a",
    },
  },
};

export default theme;
