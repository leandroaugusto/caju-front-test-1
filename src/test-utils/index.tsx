import { ReactElement } from "react";
import { ThemeProvider } from "styled-components";
import { render, RenderOptions, RenderResult } from "@testing-library/react";

import { ITestUtils } from "./types";

import theme from "~/theme";

const AllTheProviders: React.FC = ({ children }: ITestUtils) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult =>
  render(ui, {
    wrapper: AllTheProviders,
    ...options,
  });

export * from "@testing-library/react";

export { customRender };
