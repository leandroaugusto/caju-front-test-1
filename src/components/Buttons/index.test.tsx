import { faker } from "@faker-js/faker";

import Button, { ButtonSmall } from ".";
import { IconButton } from "./IconButton";
import { customRender, screen } from "~/test-utils";

import theme from "~/theme";

const MockSvgIcon = () => (
  <img data-testid="icon" src={faker.image.avatar()} alt="icon" />
);

describe("Button", () => {
  it("Should render button", () => {
    const { container } = customRender(<Button>Ativar</Button>);
    const button = screen.getByRole("button");

    expect(button).toHaveTextContent("Ativar");
    expect(button).toHaveStyle("background-color: #64a98c");
    expect(container).toMatchSnapshot();
  });

  it("Should render button with another child text", () => {
    customRender(<Button data-testid="button">Aprovar</Button>);
    const button = screen.getByRole("button");

    expect(button).toHaveTextContent("Aprovar");
  });

  it("Should render small button without theme", () => {
    const { container } = customRender(
      <ButtonSmall data-testid="button">Aprovar</ButtonSmall>
    );
    const button = screen.getByTestId("button");
    expect(button).toHaveTextContent("Aprovar");
    expect(button).toHaveStyle(`background-color: ${theme.colors.primary}`);
    expect(container).toMatchSnapshot();
  });

  it("Should render small button with specific theme", () => {
    customRender(
      <ButtonSmall data-testid="button" $buttonTheme="approved">
        Aprovar
      </ButtonSmall>
    );

    const button = screen.getByTestId("button");
    expect(button).toHaveTextContent("Aprovar");
    expect(button).toHaveStyle(
      `background-color: ${theme.colors.buttons.approved}`
    );
  });

  it("Should render icon button", () => {
    customRender(
      <IconButton data-testid="button">
        <MockSvgIcon />
      </IconButton>
    );
    const icon = screen.getByTestId("icon");

    expect(icon).toBeInTheDocument();
  });
});
