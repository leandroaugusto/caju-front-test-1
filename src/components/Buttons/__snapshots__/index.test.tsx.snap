// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > Should render button 1`] = `
<div>
  <button
    class="sc-braxZu etJSfF"
  >
    Ativar
  </button>
</div>
`;

exports[`Button > Should render button with another child text 1`] = `
<div>
  <button
    class="sc-braxZu etJSfF"
    data-testid="button"
  >
    Aprovar
  </button>
</div>
`;

exports[`Button > Should render button with specific theme 1`] = `
<div>
  <button
    class="sc-braxZu etJSfF"
    data-testid="button"
  >
    Aprovar
  </button>
</div>
`;

exports[`Button > Should render icon button 1`] = `
<div>
  <button
    class="sc-elDIKY feEgea"
    data-testid="button"
  >
    <img
      alt="icon"
      data-testid="icon"
      src="https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/1173.jpg"
    />
  </button>
</div>
`;

exports[`Button > Should render small button with specific theme 1`] = `
<div>
  <button
    class="sc-gJhJTp hEpyQs"
    data-testid="button"
  >
    Aprovar
  </button>
</div>
`;

exports[`Button > Should render small button without theme 1`] = `
<div>
  <button
    class="sc-gJhJTp ftnevY"
    data-testid="button"
  >
    Aprovar
  </button>
</div>
`;

exports[`Button Should render button 1`] = `
<div>
  <button
    class="sc-braxZu etJSfF"
  >
    Ativar
  </button>
</div>
`;

exports[`Button Should render button with specific theme 1`] = `
<div>
  <button
    class="sc-braxZu etJSfF"
    data-testid="button"
  >
    Aprovar
  </button>
</div>
`;

exports[`Button Should render small button without theme 1`] = `
<div>
  <button
    class="sc-braxZu etJSfF"
    data-testid="button"
  >
    Aprovar
  </button>
</div>
`;
